File: stringBasedProblems/problem_4.py
# Problem 4 : How do you find the maximum occurring character in a given string?

text = input("Enter your String: ")

index=0
count=0
lastHeighestValue=0

for x in range(0,len(text)):
    for y in range(x+1,len(text)):
        if(text[x]==text[y]):
            count+=1
    if(count>lastHeighestValue):
        lastHeighestValue=count
        count=0
        index=x

if(lastHeighestValue==0):
    print("There are no maximum occurring character in the given string")
else:
 print(f"Maximum occurring character in the given string is {text[index]}")

File: stringBasedProblems/problem_4.js
// Problem 4 : How do you find the maximum occurring character in a given string?

function myFunc(text){
    index = 0
    count=0
    lastHeighestValue=0

var arrayOfChar = text.split('')
for(var i=0;i<arrayOfChar.length;i++){
    for(var j=i+1;j<arrayOfChar.length;j++){
        if(arrayOfChar[i]==arrayOfChar[j]){
            count++
        }
    }
    if(count>lastHeighestValue){
        lastHeighestValue = count
        index = i
    }
    count = 0
}
if(lastHeighestValue==0){
    console.log(("There are no maximum occurring character in the given string"))
}else{
    console.log((`There is ${lastHeighestValue+1} ${arrayOfChar[index]} char in the given string`))
}
}

myFunc("auntorttt")